arch/arm/src/samv7/Kconfig

```diff
@@ -3786,6 +3786,19 @@ config SAMV7_TC3_QE
 	---help---
 		Reserve TC3 for use by QEncoder.
 
+config SAMV7_QENCODER_ENABLE_GETINDEX
+	bool "Support ioctl GETINDEX call"
+	default n
```

Comment:
@zdebanos maybe we can define this GETINDEX ioctl() as standard, instead of adding this new CONFIG, what do you think?

Reply:
@acassis The problem with making this a standard, in my opinion, is that this configuration of SAMV7's timer is not the way Microchip recommends in their datasheet.

That doesn't mean this solution does not work, in the end this solution works much better, because the driver also implements range extension up to 32 bits. The default recommended way zeroes the timer when the index is hit.

I think it could have been possible to do the calibration with the default Timer's settings (polling a status flag in the register should do the thing), however the position measurement would be much harder. I suppose this peripheral could be handled properly if it were for a custom bare-metal API, however with NuttX, this approach is much easier, in my opinion.

If it becomes the standard, it will not comply with the Timer's default settings. However, if someone wanted to use the encoder the way it is intended, it should be possible.

Reply:
The other problem is that there can exist code which uses original behavior when index resets the counter. I consider it as impractical for any serous control task but I can imagine some special cases when it can be used that way.

Reply:
In the fact there can be two independent options configurable for each encoder, the first index resets the counter, the second the position and index should be extended to 32-bits. But when you consider adding these 8 options and all ifdefs etc. then we have decided to leave single option and simpler code for now. When somebody know what is he/she doing when enables GETINDEX on SAMv7 then all previous behavior can be achieved back by subtraction index from position, limiting size etc.